### PR TITLE
MacOs build and install issues

### LINKED ISSUE #1326: MacOs issues
## Problem

1) We need to add build location (place where `openpype_console` is to sys path.
2) There can be issue during build with multiple versions of the same file (`liblzma.dynlib`) for me being copied to the same directory). This is know bug in **cx_freeze** and is already fixed and merged - marcelotduarte/cx_Freeze#915 - but not released yet. In that case we need to add it to `pyproject.toml` as git dependency pointing to master head.
3) Igniter UI has wrong scale of fonts
4) When Igniter is creating OpenPype version from frozen code it will not include `repos` for some reason. Result is error like this when running OpenPype:

image

5) When module fails to import - like **OpenTimelineIO** - it will produce absolutely cryptic and misleading traceback:

image

6) Running from sources with `poetry run python start.py tray --debug` will still result in error:

image

It seems we are importing **avalon** but not having it in `sys.path`

All tested on macos catalina 10.15.

### PR DESCRIPTION
**Pull request by mkolar**
This resolves #1326 well most of it apart from cosmetics in igniter.

Also added are:
- mac now creates OpenPype.app with nice icon
- .dmg installer is generated if the building machine has create-dmg tool in path. 

create-dmg can be installed using homebrew. this also sorts out some issues for linux coincidentally

I've also re-generated linux OTIO wheel to correctly include all the files. We are now only missing otio potentially for macOS 10. @antirotor you're the only one that has functional Catalina I think. could you make it please?

### CODE REVIEW

**Review by iLLiCiTiT — APPROVED** [review]
Looks good

**Comment by antirotor** [comment]
📧 correct OpenTimelineIO for macosx 10.15 (Catalina) is uploaded to openpype.io

File: tools/build.sh [hunk 1]
```diff
@@ -158,9 +158,26 @@ main () {
...
+
+if command -v create-dmg > /dev/null 2>&1; then
+    create-dmg \
+      --volname "OpenPype Installer" \
+      --window-pos 200 120 \
+      --window-size 600 300 \
+      --app-drop-link 100 50 \
+      "$openpype_root/build/OpenPype-Installer.dmg" \
+      "$openpype_root/build/OpenPype.app"
+  elseß
```
Comment (antirotor) [inline tools/build.sh:177]: there is some typo in `else`

**Review by antirotor — COMMENTED** [review]
On Catalina it almost work (except that typo). It has some problems during dmg creation:

image
but that can be connected with virtualized system as it is incredibly slow. It produces file named `rw.OpenPype-installer.dmg`. But installation (copying to Applications) works.  "Run without installation" doesn't work - but currently it probably doesn't work anywhere. "Install" works, but OpenPype will still not run.

If I try to run it from Terminal to see whats wrong, I've got:
```
LSOpenURLsWithRole() failed with error -10810 for the file /Applications/OpenPype.app.
```

Running `openpype_gui` directly in application bundle results in:
image

**Comment by antirotor** [comment]
That `invalid directory error` is present on Ubuntu also, so I think that is nothing mac specific

**Comment by mkolar** [comment]
> That invalid directory error is present on Ubuntu also, so I think that is nothing mac specific

@antirotor the problem is at calculation of the version path. I believe it might actually be the space in it, which is no properly handled

**Comment by antirotor** [comment]
> I believe it might actually be the space in it

but on linux, there is no space in path. I'll try to debug it. Also `create-dmg` command check should run only on macos and not linux.

**Comment by antirotor** [comment]
Ok, the problem is caused by missing `repos` in installed version.

**Comment by mkolar** [comment]
> we should check for them before build.

Definitely. can you ammend this PR?

**Comment by 64qam** [comment]
Working fine on Catalina and virtual Ubuntu.

**Comment by antirotor** [comment]
Just note - if you ever tried to build it on system with incomplete opentimelineio wheels, you need to purge poetry cache, otherwise correct wheel won't be downloaded. Run `poetry cache clear openpype --all` to achieve this.

**Pull request merged**